docs/source/architecture.rst

```diff
@@ -11,7 +11,41 @@ The system architecture is shown in the figure below, illustrating the core proc
 Natural Language Understanding
 ------------------------------
 
-The :py:class:`NLU <moviebot.nlu.nlu>` component converts the natural language :py:class:`UserUtterance <moviebot.core.utterance.utterance.UserUtterance>` into a :py:class:`DialogueAct <moviebot.dialogue_manager.dialogue_act>`. This process, comprising of *intent detection* and *slot filling*, is performed based on the current dialogue state.
+The :py:class:`NLU <moviebot.nlu.nlu>` component converts the natural language :py:class:`UserUtterance <moviebot.core.utterance.utterance.UserUtterance>` into a :py:class:`DialogueAct <moviebot.dialogue_manager.dialogue_act>`. This process, comprising of *intent detection* and *slot filling*, is performed based on the current dialogue state. The component offers two distinct modules: Rule-Based and Neural using JointBERT.
+
+Rule-Based NLU
+^^^^^^^^^^^^^^
+
+The rule-based NLU module utilizes a combination of keyword extraction and heuristics to determine the user's intent and extract slot-value pairs from their utterances. This approach relies on predefined rules and patterns to interpret user input.
+
+Neural NLU with JointBERT
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Neural NLU module employs JointBERT, a neural model trained specifically for intent detection and slot filling tasks. JointBERT predicts both the intent of the user's utterance and the corresponding slot-value pairs using a trained model.
+
+Training the JointBERT Model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To train the JointBERT model, the provided training script (`/Users/2920807/Repos/moviebot/moviebot/nlu/annotation/joint_bert/joint_bert_train.py`) can be utilized. This script fine-tunes the pre-trained BERT model on a dataset annotated with intents and slot-value pairs. Below is an overview of the training process:
```

Comment:
Does the script download the base BERT model or is that needed as "step 0"?

Reply:
It downloads the pre-trained BERT model automatically.

Reply:
Then just mention it: "This script downloads a pre-trained BERT model and fine-tunes it on a dataset..."